lib/std/time.zig

```diff
@@ -344,6 +344,221 @@ test Timer {
     try testing.expect(timer.read() < time_1);
 }
 
+pub fn Time(precision_: comptime_int, comptime has_offset: bool) type {
+    const multiplier: comptime_int = try std.math.powi(usize, 10, precision_);
+    return struct {
+        hour: Hour = 0,
+        minute: Minute = 0,
+        /// Allows leap seconds.
+        second: Second = 0,
+        /// Milliseconds, microseconds, or nanoseconds.
+        subsecond: Subsecond = 0,
+        /// Offset of `hour` and `minute` from in minutes from UTC.
+        offset: IMinutes = 0,
```

Comment:
two more comments:
- IMHO, an offset only makes sense in combination with a date since essentially, offsets originates from timezones, of which the rules are date- (and time-) dependent. My suggestion would be to move this field to `DateTime`
- my other suggestion would be to represent the offset in *seconds* (East of Greenwich), as [TZif](https://datatracker.ietf.org/doc/html/rfc8536) does it